src/Arcus.Messaging.Pumps.ServiceBus/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -246,5 +277,19 @@ private static void AddServiceBusMessagePump<TMessagePump>(IServiceCollection se
             services.AddSingleton(serviceProvider => new AzureServiceBusMessagePumpSettings(entityName, subscriptionName, getConnectionStringFromConfigurationFunc, getConnectionStringFromSecretFunc, messagePumpOptions, serviceProvider));
             services.AddHostedService<TMessagePump>();
         }
+
+        /// <summary>
+        ///     Adds a <see cref="IAzureServiceBusMessageHandler{TMessage}"/> implementation to process the messages from Azure Service Bus resources.
+        /// </summary>
+        /// <typeparam name="TMessageHandler">The type of the implementation.</typeparam>
+        /// <typeparam name="TMessage"></typeparam>
+        /// <param name="services">The collection of services to use in the application.</param>
+        public static IServiceCollection WithMessagePumpHandler<TMessageHandler, TMessage>(this IServiceCollection services)
+            where TMessageHandler : class, IAzureServiceBusMessageHandler<TMessage>
```

Comment:
This should not enforce `IAzureServiceBusMessageHandler` but `IMessageHandler` instead, otherwise there is no point in the decouple.

Imagine we add event hubs, I should have the option to use the specific interface for that one or use `IMessageHandler `but with less specific integration such as message context.

Reply:
Yes, right, but for the `AzureMessagePump`, that message pump can only use message handlers with a specific context. So, for that pump, we should only register those kind of implementations.

Maybe we should even change the signature to reflect this. `.WithAzureServiceBusMessageHandler`.

Reply:
We could maybe provide both. Since indeed, we could also register message handlers in a different context. But those will not be selected when processing messages in a Service Bus message pump.
That extension should probably be in another namespace too.

Reply:
Question is, why shouldn't we be allowed to use the generic message handlers? if I don't need the service bus message context and regular context is fine, it should be possible, no?